src/aat/resources/features/F-045/F-045.feature

```diff
@@ -0,0 +1,65 @@
+    When a request is prepared with appropriate values
+    And the request [has a valid case-reference in input parameters]
+    And it is submitted to call the [Grant access to case] operation of [CCD Data Store]
+    Then a positive response is received
+    And the response [has the 201 return code]
+    And the response has all the details as expected
+
+  @S-152
+  Scenario: must return 404 when case id is structurally valid but not exist in CCD
+    Given a user with [a detailed profile in CCD]
+    When a request is prepared with appropriate values
+    And the request [contains structurally valid but non existing case-reference in input parameters]
+    And it is submitted to call the [Grant access to case] operation of [CCD Data Store]
+    Then a negative response is received
+    And the response [has the 404 return code]
+    And the response has all the details as expected
+
+  @S-153
+  Scenario: must return negative response when request does not provide valid authentication credentials
+    Given a user with [a detailed profile in CCD]
+    When a request is prepared with appropriate values
+    And the request [does not provide the valid authentication credentials]
+    And it is submitted to call the [Grant access to case] operation of [CCD Data Store]
+    Then a negative response is received
+    And the response [has the 403 return code]
+    And the response has all the details as expected
+
+  @S-154
+  Scenario: must return negative response when request provides authentic credentials without authorized access to the operation
+    Given a user with [a detailed profile in CCD]
+    When a request is prepared with appropriate values
+    And the request [does not provide the authorised access to the operation]
+    And it is submitted to call the [Grant access to case] operation of [CCD Data Store]
+    Then a negative response is received
+    And the response [has the 403 return code]
+    And the response has all the details as expected
+
+  @S-544
+  Scenario: must return negative response when request body doesn't provide the mandatory field
+    Given a user with [a detailed profile in CCD]
+    When a request is prepared with appropriate values
+    And the request [does not provide the mandatory field for the operation]
+    And it is submitted to call the [Grant access to case] operation of [CCD Data Store]
+    Then a negative response is received
+    And the response [has the 400 return code]
+    And the response has all the details as expected
+
+  @S-545
+  Scenario: must return negative response when case id contains some non-numeric characters
+    Given a user with [a detailed profile in CCD]
+    When a request is prepared with appropriate values
+    And the request [does not provide the numeric case id for the operation]
+    And it is submitted to call the [Grant access to case] operation of [CCD Data Store]
+    Then a negative response is received
+    And the response [has the 500 return code]
```

Comment:
This must not return a 500. It seems it is, but then we should open a defect, and link it to this ticket. And expect a 400. The scenario should be disabled until the fix is available.

Reply:
Sure, I will open a defect and put @Ignore to disable this scenario.

Reply:
Please put the RDM-XYZ number of the defect ticket to here, for traceability.
And the expected response code should ideally be 400. The dest script and data should be correct, and by that way, the test scenario and data will fail as per the current function, because the current function is defective.

Reply:
Please also copy-paste the final content of the test scenario in the jira ticket for convenience.

Reply:
The JIRA content is updated with 500 response. JIRA Link is : https://tools.hmcts.net/jira/browse/RDM-6694